tests/unittests/tests-fib/tests-fib.c

```diff
@@ -0,0 +1,560 @@
+*/
+static void test_fib_13_get_next_hop_fail_on_buffer_size(void)
+{
+    size_t add_buf_size = 16; /* includes space for terminating \0 */
+    char addr_dst[] = "Test address 13";
+    kernel_pid_t iface_id = KERNEL_PID_UNDEF;
+    uint32_t next_hop_flags = 0;
+    size_t add_buf_size_nxt = 12;
+    char addr_nxt[add_buf_size];
+
+    size_t entries = 20;
+    _fill_FIB_multiple(entries, 11);
+    TEST_ASSERT_EQUAL_INT(20, fib_get_num_used_entries());
+    TEST_ASSERT_EQUAL_INT(20, universal_address_get_num_used_entries());
+
+    int ret = fib_get_next_hop(&iface_id,
+                               (uint8_t *)addr_nxt, &add_buf_size_nxt, &next_hop_flags,
+                               (uint8_t *)addr_dst, add_buf_size - 1, 0x13);
+
+    TEST_ASSERT_EQUAL_INT(-ENOBUFS, ret);
+    TEST_ASSERT_EQUAL_INT(add_buf_size_nxt, add_buf_size - 1);
+
+#if (TEST_FIB_SHOW_OUTPUT == 1)
+    fib_print_fib_table();
+    puts("");
+    universal_address_print_table();
+    puts("");
+#endif
+    fib_deinit();
+}
+
+/*
+* @brief testing prefix and exact match
+* It is expected receive 23 for addr123 as exact match and
+* 12 for addr124
+*/
+static void test_fib_14_exact_and_prefix_match(void)
+{
+    size_t add_buf_size = 16;
+    char addr_dst[add_buf_size];
+    char addr_nxt[add_buf_size];
+    kernel_pid_t iface_id = KERNEL_PID_UNDEF;
+    uint32_t next_hop_flags = 0;
+    char addr_lookup[add_buf_size];
+    memset(addr_dst, 0, add_buf_size);
+    memset(addr_nxt, 0, add_buf_size);
+
+    snprintf(addr_dst, add_buf_size, "Test addr12");
+    snprintf(addr_nxt, add_buf_size, "Test address %02d", 12);
+    fib_add_entry(42, (uint8_t *)addr_dst, add_buf_size - 1, 0x12,
+                  (uint8_t *)addr_nxt, add_buf_size - 1, 0x12, 100000);
+
+    snprintf(addr_dst, add_buf_size, "Test addr123");
+    snprintf(addr_nxt, add_buf_size, "Test address %02d", 23);
+    fib_add_entry(42, (uint8_t *)addr_dst, add_buf_size - 1, 0x123,
+                  (uint8_t *)addr_nxt, add_buf_size - 1, 0x23, 100000);
+
+    snprintf(addr_dst, add_buf_size, "Test addr1234");
+    snprintf(addr_nxt, add_buf_size, "Test address %02d", 34);
+    fib_add_entry(42, (uint8_t *)addr_dst, add_buf_size - 1, 0x1234,
+                  (uint8_t *)addr_nxt, add_buf_size - 1, 0x34, 100000);
+
+    /* exact match */
+    snprintf(addr_lookup, add_buf_size, "Test addr123");
+    int ret = fib_get_next_hop(&iface_id,
+                               (uint8_t *)addr_nxt, &add_buf_size, &next_hop_flags,
+                               (uint8_t *)addr_lookup, add_buf_size - 1, 0x123);
+
+    TEST_ASSERT_EQUAL_INT(0, ret);
+
+    char addr_expect_01[] = "Test address 23";
+    ret = strncmp(addr_expect_01, addr_nxt, add_buf_size - 1);
+    TEST_ASSERT_EQUAL_INT(0, ret);
+
+    /* prefix match */
+    add_buf_size = 16;
+    memset(addr_nxt, 0, add_buf_size);
+    memset(addr_lookup, 0, add_buf_size);
+
+    /* cppcheck: addr_lookup is only passed but not required to be read, 
```

Comment:
I think github tricks us, inspecting the file in [raw](https://raw.githubusercontent.com/BytesGalore/RIOT/add_fib/tests/unittests/tests-fib/tests-fib.c) shows the whitespace is gone :)